src/org/pentaho/reporting/platform/plugin/async/SimpleDelegatedListenableFuture.java

```diff
@@ -35,17 +36,19 @@
 
   private ListenableFuture<T> delegate;
 
+  private AtomicBoolean canceled = new AtomicBoolean( false );
+
   public SimpleDelegatedListenableFuture( final ListenableFuture<T> delegate ) {
     ArgumentNullException.validate( "delegate", delegate );
     this.delegate = delegate;
   }
 
-  @Override public boolean cancel( final boolean mayInterruptIfRunning ) {
-    return delegate.cancel( mayInterruptIfRunning );
+  @Override public synchronized boolean cancel( final boolean mayInterruptIfRunning ) {
```

Comment:
If you are using an AtomicBoolean (and dont do anything else), you no longer need the synchronized keyword here.